src/relayserver/RegistrationManager.ts

```diff
@@ -124,19 +135,38 @@ export class RegistrationManager {
           break
       }
     }
-    const wasRegistered = await this._wasRegistered()
-    if (!wasRegistered || forceRegistration) {
-      receipts = receipts.concat(await this.attemptRegistration())
+    const isRegistrationCorrect = await this._isRegistrationCorrect(hubEventsSinceLastScan)
+    if (!isRegistrationCorrect || forceRegistration) {
+      receipts = receipts.concat(await this.attemptRegistration(hubEventsSinceLastScan))
     }
     return {
       receipts,
       unregistered
     }
   }
 
-  async _wasRegistered (): Promise<boolean> {
-    const registrationBlock = await this._getRegistrationBlock()
-    return registrationBlock !== 0
+  async _isRegistrationCorrect (hubEventsSinceLastScan: EventData[]): Promise<boolean> {
+    const lastRegisteredTxSinceLastScan = reduceToLatestTx(hubEventsSinceLastScan.filter((it) => it.event === RelayServerRegistered))
+    if (lastRegisteredTxSinceLastScan != null) {
+      this.lastMinedRegisterTransaction = lastRegisteredTxSinceLastScan
+    }
+    if (this.lastMinedRegisterTransaction == null) {
+      this.lastMinedRegisterTransaction = await this._findOlderRegistrationEvent()
+    }
+    return isRegistrationValid(this.lastMinedRegisterTransaction, this.config, this.managerAddress)
+  }
+
+  async _findOlderRegistrationEvent (): Promise<EventData | undefined> {
```

Comment:
is it older ? you're using reduceToLatest, which is the youngest event..

Reply:
renamed to '_queryLatestXX'